ddl/ddl_api.go

```diff
@@ -178,6 +178,19 @@ func buildColumnsAndConstraints(ctx context.Context, colDefs []*ast.ColumnDef,
 	return cols, constraints, nil
 }
 
+func buildColumns(colNames []string) []*table.Column {
+	var cols []*table.Column
+	for i, colName := range colNames {
+		col := table.ToColumn(&model.ColumnInfo{
```

Comment:
MySQL also holds the types of every view's column. 
```
mysql> desc v1;
+-------+---------+------+-----+---------+-------+
| Field | Type    | Null | Key | Default | Extra |
+-------+---------+------+-----+---------+-------+
| a     | int(11) | YES  |     | NULL    |       |
+-------+---------+------+-----+---------+-------+
1 row in set (0.00 sec)

mysql> show create table v1;
+------+------------------------------------------------------------------------------------------------------------------------+----------------------+----------------------+
| View | Create View                                                                                                            | character_set_client | collation_connection |
+------+------------------------------------------------------------------------------------------------------------------------+----------------------+----------------------+
| v1   | CREATE ALGORITHM=UNDEFINED DEFINER=`root`@`localhost` SQL SECURITY DEFINER VIEW `v1` AS select `t`.`a` AS `a` from `t` | utf8                 | utf8_general_ci      |
+------+------------------------------------------------------------------------------------------------------------------------+----------------------+----------------------+
1 row in set (0.00 sec)

```

Reply:
MySQL does not actually holds the type, but inferred from the source table. For example, if you drop the column `t.a`, then add another column `t.a` but with type `char`, then the type of `v1.a` will be automatically changed.

Reply:
Ok~